src/learning-catalogue/validator/validator.ts

```diff
@@ -34,6 +34,14 @@ export class Validator<T> {
 		this._factory = value
 	}
 
+	get eventFactory() {
```

Comment:
why do we need a getter and setter here? Factory is passed as a constructor parameter - do we ever need to retrieve it later?